test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTrimTest.java

```diff
@@ -183,6 +191,63 @@ public void testCheckpointTrimDuringPlayback() throws Exception {
 
         // Sync should encounter trim exception, reset, and use checkpoint
         assertThat(newTestMap)
-                 .containsKeys("a", "b", "c");
+                .containsKeys("a", "b", "c");
+    }
+
+    /**
+     * Test that prefixTrim is retried in the event of a client with the wrong epoch.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testTrimRetryServerEpochChange() throws Exception{
+        final int nEntries = 5;
+        final long initEpoch = 0;
+        final long finalEpoch = 1;
+
+        // Initialize map.
+        Map<String, String> testMap = getDefaultRuntime().getObjectsView().build()
+                .setTypeToken(new TypeToken<SMRMap<String, String>>() {
+                })
+                .setStreamName("test")
+                .open();
+
+        // Populate map with 5 entries.
+        testMap.put("k1", "1");
```

Comment:
I think you can omit writing, just exercise the ::prefixTrim ?

Reply:
Done